garak/detectors/base.py

```diff
@@ -149,19 +151,21 @@ def detect(self, attempt: garak.attempt.Attempt) -> List[float | None]:
                 logging.critical(
                     "Exception running %s" % self.__class__.__name__, exc_info=e
                 )
-                return []
+                return [None] * len(all_outputs)
             else:
                 raise Exception() from e
 
-        # detector_results = [None] * len(attempt.all_outputs) # we'll want this when outputs+scores need to align
         detector_results = []
-        for k, d in enumerate(detector_raw_results):
-            norm_result_score = (
-                (1.0 - d["score"]) / 2
-                if d["label"] != self.detector_target_class
-                else (1.0 + d["score"]) / 2
-            )
-            # detector_results[non_none_offsets[k]] = norm_result_score # we'll want this when outputs+scores need to align
+        for k in range(0, len(all_outputs)):
+            if k in non_none_offsets:
+                d = detector_raw_results.pop()
```

Comment:
`pop()` takes the last item but `k` is moving forward, is this the right order? do we have tests that can catch this?